SalesforceCommerceCloud/pwa-kit
# Extensibility - Overrides fixes

# Description
This PR aims to fix 2 issues:
1) If the `extends` property in package.json contains a namespace, currently the project becomes unable to extend.

This PR modifies our override plugin's logic to handle extends that contain a namespace

2) If a directory in overrides currently contains a file that shares a name with a .test or .mock file, the overrides prioritize reading from the .test or .mock file

For example:
/app/home/index.js
/app/home/index.test.js

Details:
In the overrides plugin's file hashmap, both index.js and index.test.js currently map to the same key, which causes a collision. This PR aims to update the override plugin so that .test and .mock are indexed separately in the hashmap.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR

- Generate an extensible app
- Copy template-retail-react-app's /app directory into the generated project's overrides/app directory
- Start the generated project

Expected results:
App starts with no issues

Actual results:
Lots of warnings and errors about missing properties as the import tries to import from the .test file. For example:
```
  WARNING in ./node_modules/retail-react-app/app/components/social-icons/index.jsx 54:10-27
  export 'SocialTwitterIcon' (imported as 'SocialTwitterIcon') was not found in 'retail-react-app/app/components/icons' (module has no exports)
   @ ./overrides/app/components/social-icons/index.test.js 8:0-77 13:58-69 19:58-69
   @ ./node_modules/retail-react-app/app/components/footer/index.jsx 14:0-71 194:65-76
   @ ./overrides/app/components/_app/index.jsx 34:0-60 315:65-71
   @ ./node_modules/pwa-kit-react-sdk/ssr/server/react-rendering.js 17:34-73
```

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
